crates/subspace-archiving/src/archiver.rs

```diff
@@ -676,16 +676,40 @@ impl Archiver {
 
                 record_part.copy_from_slice(shard_chunk);
                 witness_part.copy_from_slice(
-                    &merkle_tree
+                    &records_merkle_tree
                         .get_witness(position)
                         .expect("We use the same indexes as during Merkle tree creation; qed"),
                 );
             });
 
+        let mut object_mappings_leafs = object_mapping
+            .iter()
+            .flat_map(|PieceObjectMapping { objects }| objects)
+            .map(|PieceObject::V0 { hash, offset }| {
+                hash.iter()
+                    .copied()
+                    .chain(offset.to_be_bytes())
+                    .collect::<Vec<_>>()
+            });
+
+        let object_mappings_root = match object_mapping
+            .iter()
+            .map(|PieceObjectMapping { objects }| objects.len())
+            .sum::<usize>()
+        {
+            0 => Default::default(),
+            1 => object_mappings_leafs
+                .next()
+                .map(|item| crypto::sha256_hash(item.as_ref()))
+                .expect("Has at least one item"),
+            _ => MerkleTree::from_data(object_mappings_leafs).root(),
+        };
```

Comment:
That merkle tree is a specific version designed for records, it is not suitable for other purposes and you'll not be able to verify it without knowing index in the tree.

Reply:
Could you maybe point me to some specific merkle tree which I can use here?

Reply:
There isn't any specific yet, I need to think hard about what we want here, that is why description of the issue ends with "need to figure that out".